test/jdk/javax/swing/JTable/PrintAllPagesTest.java

```diff
@@ -0,0 +1,160 @@
+
+    public static void main(String[] args) throws Exception {
+
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                createUI();
+                printAllPagesTest();
+            });
+
+            Thread.sleep(1000);
+            SwingUtilities.invokeAndWait(() -> {
+                try {
+                    table.print();
+                } catch (PrinterException e) {
+                    throw new RuntimeException("Printing failed: " + e);
+                }
+            });
+
+            // wait for latch to complete
+            boolean ret = false;
+            ret = latch.await(5, TimeUnit.MINUTES);
+
+            if (!ret) {
+                throw new RuntimeException(" User has not executed the test");
+            }
+
+            if (testResult == false) {
+                throw new RuntimeException("Only 1st page is printed out of multiple pages");
+            }
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                dispose();
+            });
+        }
+    }
+
+    private static void dispose() {
+        if (dialog != null) {
+            dialog.dispose();
+        }
+        if (f != null) {
+            f.dispose();
+        }
+    }
+
+    private static void printAllPagesTest() {
+        TableModel dataModel = new AbstractTableModel() {
+            @Override
+            public int getColumnCount() {
+                return 10;
+            }
+
+            @Override
+            public int getRowCount() {
+                return 1000;
+            }
+
+            @Override
+            public Object getValueAt(int row, int col) {
+                return Integer.valueOf(0 == col ? row + 1 : row * col);
+            }
+        };
+        table = new JTable(dataModel);
+        JScrollPane scrollpane = new JScrollPane(table);
+        table.scrollRectToVisible(table.getCellRect(table.getRowCount() - 1, 0, false));
+
+        f = new JFrame("Table test");
+        f.add(scrollpane);
+        f.setSize(1000, 800);
+        f.setLocationRelativeTo(null);
+        f.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        f.setVisible(true);
+    }
+
+    private static void createUI() {
+        String description
+                = " INSTRUCTIONS:\n"
+                + " A JTable will be shown.\n"
+                + " If only 1 page is printed,\n "
+                + " then press fail else press pass";
+
+        dialog = new JDialog();
+        dialog.setTitle("textselectionTest");
+        JTextArea textArea = new JTextArea(description);
+        textArea.setEditable(false);
+        final JButton passButton = new JButton("PASS");
+        passButton.addActionListener((e) -> {
+            testResult = true;
+            dispose();
+            latch.countDown();
+        });
+        final JButton failButton = new JButton("FAIL");
+        failButton.addActionListener((e) -> {
+            testResult = false;
+            dispose();
```

Comment:
Now that the UI is always dispose of in a finally block, the button handlers can skip the call to `dispose()`.